nselib/http.lua

```diff
@@ -87,6 +87,12 @@
 --   end
 --   </code>
 -- * <code>no_cookie_overwrite</code>: Do not overwrite the cookies initially provided and when a duplicate cookie is received, ignore it.
+-- * <code>enable_cookies</code>: Cookies are accessible to other http calls
+-- The following example code demonstrates how the cookies can be passed and updated for every subsequent http calls.
+--
+-- <code> 
+-- http.get(host, port, {no_cookie_overwrite=false, cookies = options.cookies, enable_cookie = true})
```

Comment:
Separate examples are better. In your documentation you are mixing non overwriting cookies, defining cookies and enabling them. 
So a script writer would call http.get(host, port, {enable_cookies=true}) and they would have to manually call the merge cookie function after receiving the response? Doesn't seem very transparent. The library should take care of handling the cookie automatically (and behavior can be configured)
```
first = http.get(host, port, {enable_cookies=true})
...
<here we would have to manually parse the cookies to prepare them for the next request?>
...
second = http.get(host, port, {enable_cookies=true, cookies = first.options.cookies})
```
I also think that enable_cookies should get enabled by default if you are defining a cookie string. This reduces the number of arguments needed. 

Please document this with more usage examples. Give one concrete example for each use case.